### DISCUSSION TITLE
Guidance on how to implement a text-2-sql style agent

### DISCUSSION THREAD

**Discussion by stokedout** [microsoft/teams-sdk#2168] [Q&A]
I would like to implement a text-2-sql style agent using the teams-ai library.

I want to allow users to ask questions to specific data sources, in this case a database. The difference being that the user input needs converting to SQL before seemingly calling an action that gets the data from the data source. I can see the samples talk about data sources. If I use the built in `DataSource` class then it will add the output to the prompt. Therefore I'd need to:

1. Ask a question
2. Data source takes question `temp.input` then somehow calls another prompt to get the SQL?
3. Trigger an action to get the data and render it

The above doesn't seem like a simple way forward. I'm thinking this can be achieved using a mix of LLM calls/prompts and function calls but how that maps to code in the library I'm unsure. Can you help?

I'm a bit lost with the library in general:
* Understanding augmentation types is confusing and not very well defined
* Understanding what the default prompt is meant to do is confusing
* Understanding what happens if I add multiple prompts in separate folders is not clear how it will be picked up by the planner?

**Comment by heyitsaamir** [comment]
We're clearing discussions and starting fresh. This discussion has not been acted upon. Please reopen the discussion, or start a new one if this question is still important. Thank you for your understanding.
